### ISSUE TITLE
OS X version

### ISSUE THREAD

**Issue by maksis** [airdcpp-web/airdcpp-webclient#37]
Even though the build script hasn't been fully adopted for OS X, it shouldn't require that much work to get it compiled. Having a graphical config tool for the web server might be useful though.

**Comment by knoopx** [comment]
I started writing a homebrew formula [here](https://github.com/knoopx/homebrew-airdcpp-webclient). However it produces broken binaries:
```
/usr/local/Cellar/airdcpp-webclient/2.1.0/AirDC++;Web;Client.app/Contents/ (2 files)
/usr/local/Cellar/airdcpp-webclient/2.1.0/airdcppd.app/Contents/ (2 files)
/usr/local/Cellar/airdcpp-webclient/2.1.0/share/airdcpp/ (111 files)
```

**Comment by maksis** [comment]
@knoopx do you have any idea that how it can be fixed? Note that the cmake scripts haven't been fully implemented for MacOS.

**Comment by knoopx** [comment]
No idea, make systems always looked like alien language to me. 🤣

**Comment by doobnet** [comment]
I've given this issue some thought recently. One idea would be to create a native application, with a window containing a WebKit view, i.e. an embedded browser. Inside the native application the actual client is bundled. The application starts the client as a separate process and shuts it down when the application is terminated. A preference pane is provided to choose if the browsers should connect to a remote client or the local embedded client. Some code is also added to the application to forward the notifications in the browser to native notifications.

As a second iteration there could be an option that keeps the client running although the application is terminated. When the application is started again it connects to the client.

Thoughts?

**Comment by maksis** [comment]
@doobnet Yeah, such approach sounds good to me. At a later stage, certain Web UI navigation menus could possibly be replaced with native UI elements to get some kind of native feeling in it (similar to Spotify app).

**Comment by KiPa187** [comment]
any news :) ?

**Comment by doobnet** [comment]
No, sorry. I haven't had time to work on this yet.

**Comment by TheElectrocat** [comment]
hi have a emulated clients in whinebottler mac osx high sierra, in yosemite no have problem. but have some error whit the file list big, maybe we cant solutions this.  thanks.

**Comment by TheElectrocat** [comment]
![Screen Shot 2019-08-14 at 12 49 08 AM](https://user-images.githubusercontent.com/54092782/62995172-6fcca880-be2d-11e9-8128-3893df438add.png)

**Comment by TheElectrocat** [comment]
this problem is in emulated dc++, airdc, and apexdc.

**Comment by TheElectrocat** [comment]
https://www.4shared.com/s/f_MXExGClee

**Comment by doobnet** [comment]
> hi have a emulated clients in whinebottler mac osx high sierra, in yosemite no have problem. but have some error whit the file list big, maybe we cant solutions this. thanks.

Are you running the AirDC++ Windows GUI client in Wine? I would recommend using the web client instead. The web client consists of a background daemon that is available on Linux and can be easily run from within Docker. The web frontend can be used from any platform, including macOS. I think the daemon can be compiled for macOS as well, perhaps with some minor modifications.

**Comment by doobnet** [comment]
I have started to work on a native GUI for macOS, which connects to the web client daemon. It's a very slow process and is going to take quite a while.

**Comment by TheElectrocat** [comment]
i emulated whit winebottler and the program run ok (dc++ and air dc) but the problem is (could not open targef file, general failure.) this problem is for the big file list of users, in some user all its ok and the client run very good.  thanks for your time and please work in native air dc client or emulated :) y paste the client working ok now greylink https://www.4shared.com/s/f_MXExGClee

**Comment by kaypoush** [comment]
any news on this?

**Comment by Delitants** [comment]
Hello?
